File: elementary_computer_science/Pascal/geometric_digit_number.pas
program geometric_digit_number;
uses crt;
const MN = 30;
	sup: array [1..9] of integer = (3, 2, 1, 1, 1, 1, 1, 1, 1);
var s: array [1..MN] of integer;
	n: integer;

function find: integer;
var a, q, n: integer;
begin
	n := 0;
	for a := 1 to 9 do
		for q := 1 to sup[a] do
		begin
			inc(n);
			s[n] := a * (100 + 10 * q + q * q);
		end;
	find := n;
end;

procedure display(n: integer);
var i: integer;
begin
	for i := 1 to n do write(s[i]:4);
end;

begin
	n := find;
	display(n);
	writeln;
	writeln('Total ', n, ' numbers.');
end.

File: elementary_computer_science/C#/geometric_digit_number.cs
using System;
using System.Collections;
namespace Creative1 {
	class geometric_digit_number {
		static void Main(string[] args) {
			Show(Find());
			Console.WriteLine("\n fini");
			Console.ReadLine();
		}

		static ArrayList Find() {
			ArrayList s = new ArrayList();
			int[] odd_digit = {1, 3, 5, 7, 9};
			int[] sup = {0, 3, 2, 1, 1, 1, 1, 1, 1, 1};
			for (int a = 1; a <= 9; ++a) {
				for (int q = 1; q <= sup[a]; ++q)
					s.Add(a * (100 + 10 * q + q * q));
			}
			return s;
		}

		static void Show(int[] s) {
			foreach (int x in s) Console.Write(x + " ");
		}
	}
}